From 0a171ce02a648741ea183eee819927c2d3ff64b6 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Tue, 8 Mar 2016 10:56:04 +0100
Subject: [PATCH 2/3] CMake: Install headers with PUBLIC_HEADER property on all
 platforms

This prevents the duplicate installation of header inside the framework
and in the regular include directory.
---
 CMakeLists.txt | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6d6c1b..407b901 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,7 @@ set(SIZE64 int64_t)
 configure_file(include/ogg/config_types.h.in ${CMAKE_CURRENT_SOURCE_DIR}/include/ogg/config_types.h @ONLY)
 
 set(OGG_HEADERS
+    include/ogg/config_types.h
     include/ogg/ogg.h
     include/ogg/os_types.h
 )
@@ -88,6 +89,7 @@ set_target_properties(
     IMPORT_PREFIX "lib"
     PREFIX "lib"
     SOVERSION ${OGG_VERSION_INFO}
+    PUBLIC_HEADER "${OGG_HEADERS}"
 )
 
 if(BUILD_FRAMEWORK)
@@ -98,15 +100,19 @@ if(BUILD_FRAMEWORK)
         MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${PROJECT_VERSION}
         MACOSX_FRAMEWORK_BUNDLE_VERSION ${PROJECT_VERSION}
         XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
-        PUBLIC_HEADER "${OGG_HEADERS}"
         OUTPUT_NAME Ogg
     )
 endif()
 
 configure_pkg_config_file(ogg.pc.in)
 
-install(FILES include/ogg/ogg.h include/ogg/os_types.h include/ogg/config_types.h DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/ogg)
-install(TARGETS ogg RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} FRAMEWORK DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS ogg
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    FRAMEWORK DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/ogg
+)
 if(NOT WIN32 OR CYGWIN)
     install(FILES ogg.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
-- 
2.5.0

